compiler/packages/babel-plugin-react-compiler/src/Validation/ValidateNoRefAccessInRender.ts

```diff
@@ -371,7 +371,12 @@ function validateNoRefAccessInRenderImpl(
             let lookupType: null | RefAccessType = null;
             if (objType?.kind === 'Structure') {
               lookupType = objType.value;
-            } else if (objType?.kind === 'Ref') {
+            } else if (
+              objType?.kind === 'Ref' &&
+              instr.value.kind === 'PropertyLoad' &&
+              instr.value.property === 'current'
+            ) {
+              // Only treat property access as RefValue if it's accessing the 'current' property of a ref
```

Comment:
i don't think this is quite right. we intentionally look at other properties, I think the root cause is elsewhere